cmake-proxies/cmake-modules/AudacityTesting.cmake

```diff
@@ -0,0 +1,143 @@
+if( ${_OPT}has_tests )
+   if( CMAKE_CONFIGURATION_TYPES )
+      set( TESTS_DIR "${CMAKE_BINARY_DIR}/tests" )
+      set( TESTS_DEST_DIR "${CMAKE_BINARY_DIR}/tests/${CMAKE_CFG_INTDIR}" )
+   else()
+      set( TESTS_DIR "${CMAKE_BINARY_DIR}/tests" )
+      set( TESTS_DEST_DIR "${CMAKE_BINARY_DIR}/tests/${CMAKE_BUILD_TYPE}" )
+   endif()
+
+   # Install Catch2 framework
+
+   add_conan_lib(
+      Catch2
+      catch2/2.13.8
+      REQUIRED
+      ALWAYS_ALLOW_CONAN_FALLBACK
+   )
+
+   # Setup default CTest arguments when running from IDE
+
+   set(CMAKE_CTEST_ARGUMENTS "--output-on-failure;--verbose;${CMAKE_CTEST_ARGUMENTS}")
+
+   enable_testing()
+
+   #[[
+      add_unit_test(NAME name SOURCES file1 ... LIBRARIES lib1 ...)
+
+      Creates an executable called ${name}-test from the sources files ${file1}, ... and linked
+      to libraries ${lib1}, ... Catch2 is linked implicitly.
+
+      Create a CTest test called ${name}, that expects 0 code on success
```

Comment:
... unless one uses one of the properties:
* FAIL_REGULAR_EXPRESSION
* PASS_REGULAR_EXPRESSION
* SKIP_REGULAR_EXPRESSION
* SKIP_RETURN_CODE
* WILL_FAIL

Reply:
I have a very strong opinion that unit test should return 0 on success